yoti/YotiHelper.php

```diff
@@ -292,15 +292,26 @@ public static function getFlash()
     /**
      * Generate Yoti unique username.
      *
+     * @param ActivityDetails $activityDetails
      * @param string $prefix
-     * @return string
+     *
+     * @return null|string
      */
-    private function generateUsername($prefix = 'YotiUser-')
+    private function generateUsername(ActivityDetails $activityDetails, $prefix = 'Yoti User')
     {
-        // Get the number of user_login that starts with YotiUser-
+        $givenName = $this->getUserGivenName($activityDetails);
+        $familyName = $activityDetails->getFamilyName();
+
+        // If GivenName and FamilyName are provided use as user nickname/login
+        if(!empty($givenName) && !empty($familyName)) {
+            $prefix = $givenName . " " . $familyName;
+        }
+
+        $username = $prefix = strtolower(str_replace(" ", ".", $prefix));
```

Comment:
`sanitize_user()` is used on the username at user creation time so it should be alphanumeric characters plus _ space . – * and @. I will apply a validation on the username, so if username is invalid I will use the default prefix.